packages/wallet/src/manager.ts

```diff
@@ -132,7 +136,11 @@ export class WalletManager {
     const id = wallet.getID();
     if (!this.hasWallet(id)) {
       this.walletIDs.push(id);
-      this.storage.set(StorageKey.walletListKey(), this.walletIDs);
+      try {
+        this.storage.set(StorageKey.walletListKey(), this.walletIDs);
+      } catch (error) {
+        throw new StorageError(`Failed to update wallet list: ${error}`);
```

Comment:
Here, we know for sure that storage caused the error, so wrapping it as a storage error makes sense.